bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/handler/QolsysIQZoneHandler.java

```diff
@@ -0,0 +1,139 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link QolsysIQZoneHandler} manages security zones.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class QolsysIQZoneHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(QolsysIQZoneHandler.class);
+
+    private int zoneId;
+
+    public QolsysIQZoneHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize");
+        zoneId = getConfigAs(QolsysIQZoneConfiguration.class).id;
+        initializeZone();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusChanged) {
+        logger.debug("bridgeStatusChanged {}", bridgeStatusChanged);
+        initializeZone();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    public int getZoneId() {
+        return zoneId;
+    }
+
+    protected void updateZone(Zone zone) {
+        logger.debug("updateZone {}", zone.zoneId);
+        updateState(QolsysIQBindingConstants.CHANNEL_ZONE_STATE, new DecimalType(zone.state));
+        updateZoneStatus(zone.status);
+        Map<String, String> props = new HashMap<String, String>();
+        props.put("type", zone.type);
+        props.put("name", zone.name);
+        props.put("group", zone.group);
+        props.put("zoneID", zone.id);
+        props.put("zonePhysicalType", String.valueOf(zone.zonePhysicalType));
+        props.put("zoneAlarmType", String.valueOf(zone.zoneAlarmType));
+        props.put("zoneType", zone.zoneType.toString());
+        props.put("partitionId", String.valueOf(zone.partitionId));
+        getThing().setProperties(props);
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    protected void zoneActiveEvent(ZoneActiveEvent event) {
+        if (event.zone.zoneId == getZoneId()) {
+            updateZoneStatus(event.zone.status);
+        }
+    }
+
+    protected void zoneUpdateEvent(ZoneUpdateEvent event) {
+        if (event.zone.zoneId == getZoneId()) {
+            updateZone(event.zone);
+        }
+    }
+
+    private void initializeZone() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            BridgeHandler handler = bridge.getHandler();
+            if (handler instanceof QolsysIQPartitionHandler) {
+                if (handler.getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                    return;
+                }
+                Zone z = ((QolsysIQPartitionHandler) handler).getZone(getZoneId());
+                if (z == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                    return;
+                }
+                updateZone(z);
```

Comment:
You have to set the thing status to ONLINE there.

Reply:
Sorry, it is ok, it is done in `updateZone`.

Reply:
I had that in the `updateZone` method, but it's probably cleaner to have here.